sir_cake/store/views.py

```diff
@@ -108,7 +149,10 @@ def shopping_bag(request):
         BasketItem.objects.get(id=delete_pk).delete()
         messages.success(request, "Item deleted")
 
-    basket_items = BasketItem.objects.filter(order=preorder)
+        # Delete order if it there are no itemss
+        if basket_items.count() == 0:
```

Comment:
Does this work? Shouldn't there still be a single item in `basket_items` since we deleted the last one after constructing `basket_items`? Also, I would set a `message` after deleting everything

Reply:
It works :). According to Django documentation are Querysets lazy: 

> In general, the results of a [QuerySet] aren’t fetched from the database until you “ask” for them. When you do, the QuerySet is evaluated by accessing the database.